### PR TITLE
Fix incorrect signature help trigger characters

### PR DESCRIPTION
**Pull request by insidewhy**
Fixes signature help UI in https://github.com/OmniSharp/omnisharp-roslyn/issues/2071 (sublime text) and https://github.com/ray-x/lsp_signature.nvim/issues/50 (neovim).

The VS Code plugin also currently ignores the incorrect signature help trigger characters delivered by this plugin and manually overrides them to `,` and `(`.

Fixing this will give a better out-of-the-box experience to all language-server supporting editors.

You can see here in the vs code's plugin where it's manually setting its own trigger characters:

https://github.com/OmniSharp/omnisharp-vscode/blob/master/src/omnisharp/extension.ts#L96

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/OmniSharp/omnisharp-roslyn?pullRequest=2193) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/OmniSharp/omnisharp-roslyn?pullRequest=2193) before we can accept your contribution.<br/><br/>:x: insidewhy [sign now](https://cla.dotnetfoundation.org/OmniSharp/omnisharp-roslyn?pullRequest=2193)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/OmniSharp/omnisharp-roslyn?pullRequest=2193) it.</sub>

**Review by rwols — APPROVED** [review]

**Comment by insidewhy** [comment]
@rwols This seems like a really easy and quick win, would it be possible to get this merged soon? The github rules for this project mean I have to keep the target branch up to date, so I'm constantly rebasing this against master while waiting for a review.

File: src/OmniSharp.LanguageServerProtocol/Handlers/OmniSharpSignatureHelpHandler.cs [hunk 1]
```diff
@@ -73,7 +73,7 @@ protected override SignatureHelpRegistrationOptions CreateRegistrationOptions(Si
             return new SignatureHelpRegistrationOptions()
             {
                 DocumentSelector = _documentSelector,
-                TriggerCharacters = new[] {".", "?", "["}
+                TriggerCharacters = new[] {"(", ",", "<"}
```
Comment (JoeRobich) [inline src/OmniSharp.LanguageServerProtocol/Handlers/OmniSharpSignatureHelpHandler.cs:76]: Using my highly scientific method of [trigger character discovery](https://grep.app/search?q=override%20bool%20IsTriggerCharacter&filter[repo][0]=dotnet/roslyn&filter[path.pattern][0]=SignatureHelp), I determined that `{` is needed for initializer expressions and `[` is needed for element access expressions.

```suggestion
                TriggerCharacters = new[] { "(", ",", "<", "{", "[" }
```

**Review by filipw — APPROVED** [review]
LGTM

**Pull request merged**